cli/provider/cmd.go

```diff
@@ -486,3 +482,88 @@ func (c *CmdConfigurator) ValidateFlags(ctx context.Context, cmd *cobra.Command)
 	}
 	return nil
 }
+
+func PreProcessCoverage(logger *zap.Logger, conf *config.Config) {
+	language, executable := utils.DetectLanguage(logger, conf.Command)
+	// if language is not provided through flag/config and language detected is not unknown
+	// then set the language to detected language
+	if conf.Test.Language == "" {
+		if language == models.Unknown {
+			logger.Warn("failed to detect language. please use --language to manually set the language")
```

Comment:
Mention `failed to detect language, skipping coverage caluclation. please use --language to manually set the language` and then assign true to skipping coverage. Do this for all error scenarios in preprocess.

Reply:
done